workos/session.py

```diff
@@ -0,0 +1,174 @@
+import json
+from typing import Any, Dict, List, Optional, Union
+import jwt
+from jwt import PyJWKClient
+from cryptography.fernet import Fernet
+
+from workos.types.user_management.session import (
+    AuthenticateWithSessionCookieFailureReason,
+    AuthenticateWithSessionCookieSuccessResponse,
+    AuthenticateWithSessionCookieErrorResponse,
+    RefreshWithSessionCookieErrorResponse,
+    RefreshWithSessionCookieSuccessResponse,
+)
+
+
+class SessionModule:
+    def __init__(
+        self,
+        *,
+        user_management: Any,
+        client_id: str,
+        session_data: str,
+        cookie_password: str,
+    ) -> None:
+        # If the cookie password is not provided, throw an error
+        if cookie_password is None or cookie_password == "":
+            raise ValueError("cookie_password is required")
+
+        self.user_management = user_management
+        self.client_id = client_id
+        self.session_data = session_data
+        self.cookie_password = cookie_password
+
+        self.jwks = PyJWKClient(self.user_management.get_jwks_url())
+
+        # Algorithms are hardcoded for security reasons. See https://pyjwt.readthedocs.io/en/stable/algorithms.html#specifying-an-algorithm
+        self.jwk_algorithms = ["RS256"]
+
+    def authenticate(
+        self,
+    ) -> Union[
+        AuthenticateWithSessionCookieSuccessResponse,
+        AuthenticateWithSessionCookieErrorResponse,
+    ]:
+        if self.session_data is None:
+            return AuthenticateWithSessionCookieErrorResponse(
+                authenticated=False,
+                reason=AuthenticateWithSessionCookieFailureReason.NO_SESSION_COOKIE_PROVIDED,
+            )
+
+        try:
+            session = self.unseal_data(self.session_data, self.cookie_password)
+        except Exception:
+            return AuthenticateWithSessionCookieErrorResponse(
+                authenticated=False,
+                reason=AuthenticateWithSessionCookieFailureReason.INVALID_SESSION_COOKIE,
+            )
+
+        if not session["access_token"]:
+            return AuthenticateWithSessionCookieErrorResponse(
+                authenticated=False,
+                reason=AuthenticateWithSessionCookieFailureReason.INVALID_SESSION_COOKIE,
+            )
```

Comment:
Will `access_token` always exist, or should we do `session.get("access_token", None)` for safety?